template/src/main.cpp

```diff
@@ -1,35 +1,34 @@
 #include "main.hpp"
 
-static ModInfo modInfo; // Stores the ID and version of our mod, and is sent to the modloader upon startup
+static modloader::ModInfo modInfo{MOD_ID, VERSION, 0};
```

Comment:
CModInfo and ModInfo live in different headers, Include what you use here, and also elsewhere instead of relying on main.hpp (or add the iwyu pragmas there to export them to avoid iwyu complaining about everything in this file)